### PR TITLE
Changes to js code, create set from public data

### PR DESCRIPTION
**Pull request by wilke**
Needs to be deployed on dev. 
Changes made to reflect changes in methods -> View Communities Data.

### CODE REVIEW

**Comment by wilke** [comment]
2) Explain the exact failure mode the change addresses:
Search not working - enables creation/import of metagenome sets

3) Assess the risks associated with deploying the new code:
deployment on dev for testing - not production. Widget/js code might not work in environment but no other component should be affected 

4) Explain what tests you have done to make sure the code both fixes
the issue and doesn't break anything else:

Tested on local deployment - need to test in kbase dev deployment.

**Comment by aparkin** [comment]
Ok-- I was writing just as you commented. gitHub is theoretically appropriate but we are asking for emails to Fernando and I. No worries though. I don't understand your answers to 2... how is search not working related to enabling the creation of sets?

I am ok with a dev deployment in any case.

**Comment by msneddon** [comment]
I guess I don't understand the urgency of getting this and https://github.com/kbase/narrative_method_specs/pull/181 merged and deployed into dev right now for testing if you do not plan on pushing this to production tonight.  We can wait until after next week.

Particularly for the narrative_method_specs repository, there are other changes in dev right now that we do want to go out to production tonight, so I don't want to merge anything right now that we will have to revert in the staging and master branches.  We can improve the testing workflow for code that is not planned for production in the future, but not right now.

**Comment by briehl** [comment]
I'm concerned that this includes a brand new widget with unknown functionality. I'm happy to deploy to narrative-dev, but unless this fixes some absolutely critical error that would otherwise crash the system, other than being an inconvenience, I would suggest putting off a merge to the deployment branch until after next week.

**Comment by teharrison** [comment]
This change was meant to address this issue:
https://atlassian.kbase.us/browse/NAR-491.  Namely that the metagenomes in
the 'public' tab of the data browser are of the wrong type and can't not be
used with any existing widgets.  This custom input widget gets around the
inability to use search to find metagenomes.

**Comment by aparkin** [comment]
I think we need to loop Matt into that discussion. My understanding is that
was supposed to be fixed in a search deploy.


On Thu, Feb 19, 2015 at 12:33 PM, Travis Harrison <notifications@github.com>
wrote:

> This change was meant to address this issue:
> https://atlassian.kbase.us/browse/NAR-491. Namely that the metagenomes in
> the 'public' tab of the data browser are of the wrong type and can't not be
> used with any existing widgets. This custom input widget gets around the
> inability to use search to find metagenomes.
>
> On Thu, Feb 19, 2015 at 2:15 PM, Bill Riehl <notifications@github.com>
> wrote:
>
> > I'm concerned that this includes a brand new widget with unknown
> > functionality. I'm happy to deploy to narrative-dev, but unless this
> fixes
> > some absolutely critical error that would otherwise crash the system,
> other
> > than being an inconvenience, I would suggest putting off a merge to the
> > deployment branch until after next week.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/kbase/narrative/pull/300#issuecomment-75127302>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kbase/narrative/pull/300#issuecomment-75130281>.
>



-- 
Adam Paul Arkin
Dean A. Richard Newton Memorial Chair
-------------------------------------
Director, Physical Biosciences Division
E.O. Lawrence Berkeley National Laboratory

Professor, Department of Bioengineering
University of California
Berkeley, CA, 94720

CEO/CSO, DOE Systems Biology Knowledgebase, http://kbase.us
Director, Berkeley Synthetic Biology Institute, http://synbio.berkeley.edu
PI and Co-Director, ENIGMA SFA, http://enigma.lbl.gov <http://vimss.lbl.gov>
Investigator, Energy Biosciences Institute,
http://energybiosciencesinstitute.org

Office: 512C Energy Biosciences Building (Berkeley Campus)
Mailing address:
E.O. Lawrence Berkeley National Laboratory
1 Cyclotron Road, MS 955-512L
Berkeley, CA 94720

Contact:
W: http://genomics.lbl.gov
V: 510-495-2366
C: 510-206-1389
F: 510-486-6219

Assistant:
Gwyneth A. Terry
V: 510-495-2116
E: GATerry@lbl.gov
-------------------------------------

**Comment by aparkin** [comment]
The pending search deploy includes the new metagenome types, which is what
the copy issue is in narrative, since it is using the search api and
talking to search with older data.

**Comment by aparkin** [comment]
So is the implication that Travis' fix is not necessary?

On Thu, Feb 19, 2015 at 1:12 PM, Matthew Henderson <mhenderson@lbl.gov>
wrote:

> The pending search deploy includes the new metagenome types, which is what
> the copy issue is in narrative, since it is using the search api and
> talking to search with older data.
>
>
> On Thu, Feb 19, 2015 at 12:38 PM, Adam Arkin <aparkin@lbl.gov> wrote:
>
>> I think we need to loop Matt into that discussion. My understanding is
>> that was supposed to be fixed in a search deploy.
>>
>>
>> On Thu, Feb 19, 2015 at 12:33 PM, Travis Harrison <
>> notifications@github.com> wrote:
>>
>>> This change was meant to address this issue:
>>> https://atlassian.kbase.us/browse/NAR-491. Namely that the metagenomes
>>> in
>>> the 'public' tab of the data browser are of the wrong type and can't not
>>> be
>>> used with any existing widgets. This custom input widget gets around the
>>> inability to use search to find metagenomes.
>>>
>>> On Thu, Feb 19, 2015 at 2:15 PM, Bill Riehl <notifications@github.com>
>>> wrote:
>>>
>>> > I'm concerned that this includes a brand new widget with unknown
>>> > functionality. I'm happy to deploy to narrative-dev, but unless this
>>> fixes
>>> > some absolutely critical error that would otherwise crash the system,
>>> other
>>> > than being an inconvenience, I would suggest putting off a merge to the
>>> > deployment branch until after next week.
>>> >
>>> > —
>>> > Reply to this email directly or view it on GitHub
>>> > <https://github.com/kbase/narrative/pull/300#issuecomment-75127302>.
>>> >
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/kbase/narrative/pull/300#issuecomment-75130281>.
>>>
>>
>>
>>
>> --
>> Adam Paul Arkin
>> Dean A. Richard Newton Memorial Chair
>> -------------------------------------
>> Director, Physical Biosciences Division
>> E.O. Lawrence Berkeley National Laboratory
>>
>> Professor, Department of Bioengineering
>> University of California
>> Berkeley, CA, 94720
>>
>> CEO/CSO, DOE Systems Biology Knowledgebase, http://kbase.us
>> Director, Berkeley Synthetic Biology Institute,
>> http://synbio.berkeley.edu
>> PI and Co-Director, ENIGMA SFA, http://enigma.lbl.gov
>> <http://vimss.lbl.gov>
>> Investigator, Energy Biosciences Institute,
>> http://energybiosciencesinstitute.org
>>
>> Office: 512C Energy Biosciences Building (Berkeley Campus)
>> Mailing address:
>> E.O. Lawrence Berkeley National Laboratory
>> 1 Cyclotron Road, MS 955-512L
>> Berkeley, CA 94720
>>
>> Contact:
>> W: http://genomics.lbl.gov
>> V: 510-495-2366
>> C: 510-206-1389
>> F: 510-486-6219
>>
>> Assistant:
>> Gwyneth A. Terry
>> V: 510-495-2116
>> E: GATerry@lbl.gov
>> -------------------------------------
>>
>
>


-- 
Adam Paul Arkin
Dean A. Richard Newton Memorial Chair
-------------------------------------
Director, Physical Biosciences Division
E.O. Lawrence Berkeley National Laboratory

Professor, Department of Bioengineering
University of California
Berkeley, CA, 94720

CEO/CSO, DOE Systems Biology Knowledgebase, http://kbase.us
Director, Berkeley Synthetic Biology Institute, http://synbio.berkeley.edu
PI and Co-Director, ENIGMA SFA, http://enigma.lbl.gov <http://vimss.lbl.gov>
Investigator, Energy Biosciences Institute,
http://energybiosciencesinstitute.org

Office: 512C Energy Biosciences Building (Berkeley Campus)
Mailing address:
E.O. Lawrence Berkeley National Laboratory
1 Cyclotron Road, MS 955-512L
Berkeley, CA 94720

Contact:
W: http://genomics.lbl.gov
V: 510-495-2366
C: 510-206-1389
F: 510-486-6219

Assistant:
Gwyneth A. Terry
V: 510-495-2116
E: GATerry@lbl.gov
-------------------------------------

**Comment by aparkin** [comment]
If I'm interpreting this correctly and they are trying to make sure they
handle old objects being copied to your narrative, then yes this is not
needed.

**Comment by aparkin** [comment]
I can also make the old workspace unreadable so that it is impossible to
access the old objects.

**Comment by wilke** [comment]
The widget in it's current form will be obsolete. I updated https://github.com/kbase/narrative_method_specs/pull/181 to reflect this.

**Comment by briehl** [comment]
@teharrison Oh! Okay. Roman and Matt are working on updating the Search results in the Public tab so it searches the right type. That might fix things? See here: https://github.com/kbase/narrative/pull/302

**Comment by aparkin** [comment]
When/Where will Andreas and Travis be able to test that?

On Thu, Feb 19, 2015 at 1:31 PM, Bill Riehl <notifications@github.com>
wrote:

> @teharrison <https://github.com/teharrison> Oh! Okay. Roman and Matt are
> working on updating the Search results in the Public tab so it searches the
> right type. That might fix things? See here: #302
> <https://github.com/kbase/narrative/pull/302>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kbase/narrative/pull/300#issuecomment-75140334>.
>



-- 
Adam Paul Arkin
Dean A. Richard Newton Memorial Chair
-------------------------------------
Director, Physical Biosciences Division
E.O. Lawrence Berkeley National Laboratory

Professor, Department of Bioengineering
University of California
Berkeley, CA, 94720

CEO/CSO, DOE Systems Biology Knowledgebase, http://kbase.us
Director, Berkeley Synthetic Biology Institute, http://synbio.berkeley.edu
PI and Co-Director, ENIGMA SFA, http://enigma.lbl.gov <http://vimss.lbl.gov>
Investigator, Energy Biosciences Institute,
http://energybiosciencesinstitute.org

Office: 512C Energy Biosciences Building (Berkeley Campus)
Mailing address:
E.O. Lawrence Berkeley National Laboratory
1 Cyclotron Road, MS 955-512L
Berkeley, CA 94720

Contact:
W: http://genomics.lbl.gov
V: 510-495-2366
C: 510-206-1389
F: 510-486-6219

Assistant:
Gwyneth A. Terry
V: 510-495-2116
E: GATerry@lbl.gov
-------------------------------------

**Comment by teharrison** [comment]
Bill, that would be the correct solution and make a new method / widget unnecessary.  Is that going into narrative-dev anytime soon?

**Comment by briehl** [comment]
Just chatted with Matt and Roman, and that test, pre-production endpoint is ready. I'll merge the change and redeploy narrative-dev right away.

**Comment by briehl** [comment]
Sorry for the delay, there was a last-second config change after it got deployed. You should be able to see it now on narrative-dev with version 1.0.2b.

**Comment by briehl** [comment]
If you guys agree this all works, let me know, and I'll request a narrative deploy tonight.

**Comment by mlhenderson** [comment]
"this all works" right now should be restricted to copying metagenomes from
the public data panel in narrative, and then making sure those objects work
correctly with communities methods/apps

**Comment by briehl** [comment]
I can confirm that I can use the 'Public' data panel to:
1. Search metagenomes
2. Add one to the data panel
3. Use the metagenome as an input for the "Reconstruct Community Metabolic Model" app
4. Use the metagenome as an input for the  "Create Metagenome Set" method
5. View the metagenome with the DnD viewer
6. View the metagenome collection with the DnD viewer

But others should repeat this with other metagenomes.

**Comment by briehl** [comment]
I also just pulled in a few metagenomes from the search, assembled a collection, and ran the "Build and Normalize Metagenomic Functional Abundance Data" app with it, which completed successfully.

**Comment by msneddon** [comment]
I have also tested import successfully of several new metagenome objects, confirmed that all the viewers and landing pages work with DnD and click, and generally clicked around the narrative to make sure nothing in related data panel looks broken.

**Comment by fperez** [comment]
This has a ton of new code and not a test in sight. Obviously not ready for production this late in the game.

**Comment by briehl** [comment]
I think #302 solved this problem (and was deployed) a couple weeks back. Is that true? Is it safe to close this PR now?

**Comment by kkellerlbl** [comment]
No objections, closing this old PR.
